src/orion/core/io/cmdline_parser.py

```diff
@@ -20,28 +20,39 @@
 
 
 class CmdlineParser(object):
+    """CmdlineParser is aimed at providing a simple tool to interpret commandline arguments
+    for the purposes of Orion. It can transform a list of string representing arguments to their
```

Comment:
It returns a list or a dictionary? There should be notes about specific cases like boolean arguments and array arguments. Also there should be a mention of the limitation of the parser for commandlines where there is optional arguments before a command, like it is for Oríon. For instance ```somecommand --optional argument subcommand --another argument``` would be incorrectly parsed as 
```python
{'optional': ['argument', 'subcommand'],
 'another': 'argument'}
```